services/src/main/java/org/keycloak/services/resources/admin/AdminEventBuilder.java

```diff
@@ -25,8 +34,8 @@
     private RealmModel realm;
     private AdminEvent adminEvent;
 
-    public AdminEventBuilder(RealmModel realm, KeycloakSession session, ClientConnection clientConnection) {
-        this.realm = realm;
+    public AdminEventBuilder(AdminAuth auth, KeycloakSession session, ClientConnection clientConnection) {
+        this.realm = auth.getRealm();
```

Comment:
This is incorrect - isAdminEventsEnabled and getEventsListeners should be from the realm that is being operated on, not the auth realm